### PR TITLE
#4107 Performance improvements for column sorting

### PR DESCRIPTION
**Pull request by plukasew**
#4107 

buildGradeMatrix() always sorts by user first even if the sort is for
another column. Additionally, descending sorts are first sorted in
normal order and then reversed (linear time) rather than using a
reverse-order sort to begin this.

This patch will address both issues.

### CODE REVIEW

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java [hunk 1]
```diff
@@ -683,20 +684,19 @@ public GradeSaveResponse saveGrade(final Long assignmentId, final String student
 		// this gives us our base list and will be sorted as per our desired
 		// sort method
 		final List<User> students = getUsers(studentUuids);
-		if (settings.getStudentSortOrder() != null || settings.getNameSortOrder() != null) {
```
Comment (plukasew) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/business/GradebookNgBusinessService.java]: settings.getNameSortOrder is never null, causing this block to execute and sort users by name every time

**Comment by ncaidin** [comment]
Curious to know how much performance improves after this patch? No reasonable way to test on nightly servers, AFAIK. Thanks.

**Comment by plukasew** [comment]
These performance improvements are probably not noticeable except perhaps in large sites. For testing this I would just concentrate on confirming that the sorting is still correct.

**Comment by ncaidin** [comment]
It seems to work. Sorted on Students by Last name, First name, and then by Grade point on one of the grades.
Nightly 11.x MySQL

**Pull request merged**